apps/vaporgui/NavigationEventRouter.cpp

```diff
@@ -280,6 +280,16 @@ void NavigationEventRouter::_performAutoStretching(string dataSetName) {
 	vector<double> minExt, maxExt;
 	
 	for (int i=0; i<winNames.size(); i++) {
+		ViewpointParams *vpParams = paramsMgr->GetViewpointParams(winNames[i]);
+		Transform* transform = vpParams->GetTransform(dataSetName);
+        std::vector<double> scales = transform->GetScales();
+        int zDimension = 2;
+
+        // If the z-dimension scale is not 1.f, the user has saved a session with
+        // a non-default value.  Don't modify it.
+        if ( scales[ zDimension ] != 1.f )
```

Comment:
The new code assumes that only scaling along the Z dimension is possible, while the code below is more general and supports scaling along any axis.

Reply:
That is correct.  Good to go now.